<?xml version="1.0"?>

<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:xacro="http://ros.org/wiki/xacro" name="###MODEL_NAME###">
      
      <xacro:include filename="basic/package_properties.xacro"/>

      <!-- #################################################################################### -->	
      <!-- #These are the default values for each object, it is recommended to change the here# -->	
      <!-- #################################################################################### -->	
      	      	

	  
	  <!-- if the mesh collision is prefered to the simplyfied autogenerated one -->
	  <xacro:arg name="mesh_collision" default="false" /> 
		  
	  <!-- the scaling of the object -->
	  <xacro:arg name="scale_x" default="###SCALE_X###" /> 
	  <xacro:arg name="scale_y" default="###SCALE_Y###" /> 
	  <xacro:arg name="scale_z" default="###SCALE_Z###" />	
	  
	  <!-- Inertial of the model --> 
	  <xacro:arg name="mass" 	 default="###MASS###" />	   
	  <xacro:arg name="ixx" default="###INERTIA_XX###" />	   
	  <xacro:arg name="ixy" default="###INERTIA_XY###" />	   
	  <xacro:arg name="ixz" default="###INERTIA_XZ###" />	   
	  <xacro:arg name="iyy" default="###INERTIA_YY###" />	   
	  <xacro:arg name="iyz" default="###INERTIA_YZ###" />	   
	  <xacro:arg name="izz" default="###INERTIA_ZZ###" />	   
      	
	  <xacro:property name="meshfile" default="###MESH_FILE###" />		   
<!--
	   	
       	
<link name="###OBJ_NAME###">
        <visual>
                <geometry>
        			<mesh filename="package://${package}/${mesh_dir}/${mesh_file}"/>
                </geometry>
        </visual>

        <collision>
                
        	<xacro:if value="$(arg mesh_collision)" >        		
        		<geometry>
					<mesh filename="package://${package}/${mesh_dir}/${mesh_file}"/>
                </geometry>
        	</xacro:if> 
        	<xacro:if value="${$(arg mesh_collision) == false}" >        		
        		<geometry>
					###SIMPLE_COLLISIONS###
                </geometry>
        	</xacro:if>         
        </collision>

	<inertial>
		<mass value="$(arg mass)" />
		<inertia ixx="$(arg ixx)" 
				 ixy="$(arg ixy)" 
				 ixz="$(arg ixz)"
				 iyy="$(arg iyy)"
				 izz="$(arg izz)" />
	</inertial>

</link>
       	-->		       	
       	
       	
</robot>